EDITD/kubetools
# PFM-615 / Add support to specify to ServiceAccount and SecretProviderClass

## Purpose of PR
- Add support to specify `ServiceAccount` and `SecretProviderClass` for:
   - dependencies
   - deployments
   - cronjobs

This PR assumes that `ServiceAccount` and `SecretProviderClass` are already created (if needed), before deploying the project with kubetools.

## Parts of the app this will impact
kubetools for:
   - dependencies
   - deployments
   - cronjobs

## How to tests
- Create a k8s `Namespace`
- Create a k8s `ServiceAccount` with the permissions (in the `namespace` created in step 1)
- Create a k8s `SecretProviderClass` with the list of secrets that we want to access (in the `namespace` created in step 1)
- Example of kubetools.yml file
```
name: nginx-app

deployments:
  nginx-deployment:
    serviceAccountName: <SERVICE_ACCOUNT_NAME>
    secrets:
      secret-volume:
        mountPath: <SECRETS_MOUNT_PATH>
        secretProviderClass: <SECRET_PROVIDER_CLASS_NAME>
    containers:
      nginx:
        image: nginx
        ports:
          - 80
```

- Nginx container should be up
- Exec into container and you should be able to `cat` the secret file based on the `mountPath`
